### PR TITLE
HDDS-4917.[FSO]Implement ACL requests for new layout

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
Implement ACL api's like getAcl, addAcl, setAcl ,removeAcl in the new layout V1.

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-4917

## How was this patch tested?
Added Unit test

### CODE REVIEW

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequestV1.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    return super.validateAndUpdateCache(ozoneManager, trxnLogIndex,
+        omDoubleBufferHelper);
+  }
+
+  @Override OMClientResponse onSuccess(
+      OzoneManagerProtocolProtos.OMResponse.Builder omResponse,
+      OmKeyInfo omKeyInfo, boolean operationResult, boolean isDirectory) {
+    omResponse.setSuccess(operationResult);
+    omResponse.setSetAclResponse(
+        OzoneManagerProtocolProtos.SetAclResponse.newBuilder()
```
Comment (linyiqun) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAddAclRequestV1.java:145]: Seems here should be:
```java
omResponse.setAddAclResponse(
        OzoneManagerProtocolProtos.AddAclResponse.newBuilder()
```

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyRemoveAclRequestV1.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    return super.validateAndUpdateCache(ozoneManager, trxnLogIndex,
+        omDoubleBufferHelper);
+  }
+
+  @Override OMClientResponse onSuccess(
+      OzoneManagerProtocolProtos.OMResponse.Builder omResponse,
+      OmKeyInfo omKeyInfo, boolean operationResult, boolean isDirectory) {
+    omResponse.setSuccess(operationResult);
+    omResponse.setSetAclResponse(
+        OzoneManagerProtocolProtos.SetAclResponse.newBuilder()
```
Comment (linyiqun) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyRemoveAclRequestV1.java:149]: The similar problem, expected:  
 omResponse.setRemoveAclResponse(
        OzoneManagerProtocolProtos.RemoveAclResponse.newBuilder()

Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyRemoveAclRequestV1.java]: Thanks @linyiqun for pointing this out.

**Review by linyiqun — APPROVED** [review]
LGTM, +1. Thanks for addressing the comments.
Let's wait for others to have a double check for this PR before the merge, : ).

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAclRequestV1.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        omMetadataManager.getKeyTable().addCacheEntry(new CacheKey<>(dbKey),
+            new CacheValue<>(Optional.of(omKeyInfo), trxnLogIndex));
+      }
+      omClientResponse =
+          onSuccess(omResponse, omKeyInfo, operationResult, isDirectory);
+      result = Result.SUCCESS;
+    } catch (IOException ex) {
+      result = Result.FAILURE;
+      exception = ex;
+      omClientResponse = onFailure(omResponse, ex);
```
[the remark this answers is not in the document]
Comment (sadanand48) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/acl/OMKeyAclRequestV1.java:130]: Thanks @rakeshadr for the review comment . Addressed it.

**Comment by rakeshadr** [comment]
+1 patch looks good to me.

Thanks @sadanand48 for the contribution. Thanks a lot @linyiqun for the reviews.

**Comment by rakeshadr** [comment]
@xiaoyuyao this patch is expecting prefixes and the key should exist in DB. Otw, it will throw Key_Not_Found exception. IIRC, we have discussed this case of non-existent path and not required to handle this case. Could you please add your thoughts. Thanks!

IMHO, I am open to discuss this point and if we need to support then it can be handled separately in a new jira. I'm merging this PR now.

**Pull request merged**